src/metabase/driver/postgres.clj

```diff
@@ -209,6 +213,58 @@
 
 (def ^:private ^:dynamic *enum-types* nil)
 
+(defn- get-tables-sql
+  [schemas table-names]
+  ;; Ref: https://github.com/davecramer/pgjdbc/blob/a714bfd/pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java#L1272
+  (sql/format
+   (cond-> {:select    [[:n.nspname :schema]
+                        [:c.relname :name]
+                        [[:case-expr :c.relkind
+                          [:inline "r"] [:inline "TABLE"]
+                          [:inline "p"] [:inline "PARTITIONED TABLE"]
+                          [:inline "v"] [:inline "VIEW"]
+                          [:inline "f"] [:inline "FOREIGN TABLE"]
+                          [:inline "m"] [:inline "MATERIALIZED VIEW"]
+                          :else nil]
+                         :type]
+                        [:d.description :description]]
+            :from      [[:pg_catalog.pg_namespace :n]
+                        [:pg_catalog.pg_class :c]]
+            :left-join [[:pg_catalog.pg_description :d] [:and [:= :c.oid :d.objoid] [:= :d.objsubid [:inline 0]]]
+                        [:pg_catalog.pg_class :dc]      [:and [:= :d.classoid :dc.oid] [:= :dc.relname [:inline "pg_class"]]]
+                        [:pg_catalog.pg_namespace :dn]  [:and [:= :dn.oid :dc.relnamespace] [:= :dn.nspname [:inline "pg_catalog"]]]]
+            :where     [:and [:= :c.relnamespace :n.oid]
+                        ;; filter out system tables
+                        [(keyword "!~") :n.nspname "^pg_"] [:<> :n.nspname "information_schema"]
+                        ;; only get tables of type: TABLE, PARTITIONED TABLE, VIEW, FOREIGN TABLE, MATERIALIZED VIEW
+                        [:raw "c.relkind in ('r', 'p', 'v', 'f', 'm')"]]
+            :order-by  [:type :schema :name]}
+     (seq schemas)
+     (sql.helpers/where [:in :n.nspname schemas])
+
+     (seq table-names)
+     (sql.helpers/where [:in :c.relname table-names]))
+   {:dialect :ansi}))
+
+(defn- get-tables
+  ;; have it as its own method for ease of testing
+  [database schemas tables]
+  (sql-jdbc.execute/reducible-query database (get-tables-sql schemas tables)))
+
+(defmethod driver/describe-database :postgres
+  [driver database]
+  {:tables
+   (sql-jdbc.execute/do-with-connection-with-options
+    driver
+    database
+    nil
+    (fn [^Connection conn]
+      (if-let [syncable-schemas (seq (driver/syncable-schemas driver database))]
+        (let [have-select-privilege? (sql-jdbc.describe-database/have-select-privilege-fn :postgres conn)]
+          (into #{} (comp (filter have-select-privilege?) (map #(dissoc % :type)))
+                (get-tables database syncable-schemas nil)))
```

Comment:
I think this is a little denser than idiomatic styling, which would have the `filter` and `map` steps on separate lines, similar to a `->>` expression:
```suggestion
          (into #{} 
                (comp (filter have-select-privilege?)
                      (map #(dissoc % :type)))
                (get-tables database syncable-schemas nil)))
```

Reply:
[a84ee86](https://github.com/metabase/metabase/pull/39670/commits/a84ee8604d75e92f1a180d429114cbe61ec6ff1d)